File: Projects/Dagger/Projects/Project-4/app/src/main/java/com/droid/code/di/modules/ApplicationModule.java
package com.droid.code.di.modules;

import android.content.Context;

import com.droid.code.MyApplication;
import com.droid.code.data.local.DatabaseService;
import com.droid.code.data.remote.NetworkService;
import com.droid.code.di.qualifiers.DatabaseInfo;
import com.droid.code.di.qualifiers.NetworkInfo;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;


@Module
public class ApplicationModule {

    private MyApplication application;

    public ApplicationModule(MyApplication application) {
        this.application = application;
    }

   /* @Singleton
    @Provides
    DatabaseService provideDatabaseService() {
        return new DatabaseService(application, "dummy_db", 1);
    }

    @Singleton
    @Provides
    NetworkService provideNetworkService() {
        return new NetworkService(application, "SOME_API_KEY");
    }*/

    @Singleton
    @Provides
    Context provideContext() {
        return application;
    }

    @DatabaseInfo
    @Provides
    String provideDbName() {
        return "Db-Name";
    }

    @NetworkInfo
    @Provides
    String provideNetworkInfo() {
        return "NETWORK_DUMMY_DATA";
    }

    @Singleton
    @Provides
    Integer provideDbVersion() {
        return 123;
    }




}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/di/module/ApplicationModule.kt
package com.demo.kotlin.sample.di.module

import android.content.Context

import com.demo.kotlin.sample.MyApplication
import com.demo.kotlin.sample.di.ApplicationContext
import com.demo.kotlin.sample.di.DatabaseInfo
import com.demo.kotlin.sample.di.NetworkInfo

import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val application: MyApplication) {

    @ApplicationContext
    @Provides
    fun provideContext(): Context = application

    @Provides
    @DatabaseInfo
    fun provideDatabaseName(): String = "dummy_db"

    @Provides
    @DatabaseInfo
    fun provideDatabaseVersion(): Int = 1

    @Provides
    @NetworkInfo
    fun provideApiKey(): String = "SOME_API_KEY"
}
